lib/rules/index.js

```diff
@@ -56,7 +56,13 @@ module.exports = new LazyLoadingRuleMap(Object.entries({
     "grouped-accessor-pairs": () => require("./grouped-accessor-pairs"),
     "guard-for-in": () => require("./guard-for-in"),
     "handle-callback-err": () => require("./handle-callback-err"),
-    "id-blacklist": () => require("./id-blacklist"),
+
+    /*
+     * TODO: Remove in next major release.
+     * https://github.com/eslint/eslint/issues/13407
+     */
+    "id-blacklist": () => require("./id-denylist"),
```

Comment:
I'm not sure if we can remove this rule because of [our deprecation policy](https://eslint.org/docs/user-guide/rule-deprecation).

Also, `rule.meta.deprecated` should be `true` for this rule.

Reply:
Fair point on not being able to remove the rule call-through. Will remove that comment.

In my mind, this isn't a deprecation as much as a rename, so I'm not sure it makes sense to mark it as deprecated. Since the API of the rule itself isn't changing, all existing configurations should still work.

Reply:
I think the question here is what the expected behavior of the user is. If you are expecting people to update their configurations (which seems what you're expecting given the comment about removing the rule in the future), then you need to somehow indicate to the user. The only way we have to indicate that right now is through marking the existing rule as deprecated. We can't really plan on removing a rule without warning people that it's going away.

Reply:
@mysticatea @nzakas Does removing this comment and leaving the code as-is resolve your concerns?

Reply:
For me, yes. If we don’t plan on removing the rule in the future then I think it’s fine.

Reply:
Updated!

Reply:
Yes. If you are sure that it doesn't affect our scripts for our website, I don't mind the `deprecated` flag.

Reply:
@mysticatea I think what @kaicataldo and I are saying is that we don't need to set `meta.deprecated` to `true` if we don't intend to remove it.

Reply:
I think this is the first time that a rule exists but `<rule-id>.md` doesn't exist. Therefore, my concern is the scripts about website, but I'm not familiar with the scripts. I'm OK if the scripts are OK.

Reply:
That's a valid concern - I was thinking we could just create a redirect in the website repo. Thoughts?